code/modules/mob/new_player/preferences_setup.dm

```diff
@@ -207,748 +207,71 @@
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "judge"), ICON_UNDERLAY)
-				switch(backbag)
-					if(2)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-					if(3)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel-norm"), ICON_OVERLAY)
-					if(4)
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(disabilities & NEARSIGHTED)
-		preview_icon.Blend(new /icon('icons/mob/eyes.dmi', "glasses"), ICON_OVERLAY)
-
-	// Observers get tourist outfit.
-	if(for_observer)
-		clothes_s = new /icon(uniform_dmi, "tourist_s")
-		clothes_s.Blend(new /icon('icons/mob/feet.dmi', "black"), ICON_UNDERLAY)
-		if(backbag == 2)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "backpack"), ICON_OVERLAY)
-		else if(backbag == 3 || backbag == 4)
-			clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	if(underwear_s)
-		preview_icon.Blend(underwear_s, ICON_OVERLAY)
-	if(undershirt_s)
-		preview_icon.Blend(undershirt_s, ICON_OVERLAY)
-	if(socks_s)
-		preview_icon.Blend(socks_s, ICON_OVERLAY)
-	if(clothes_s)
-		preview_icon.Blend(clothes_s, ICON_OVERLAY)
-	preview_icon.Blend(face_s, ICON_OVERLAY)
-	preview_icon_front = new(preview_icon, dir = SOUTH)
-	preview_icon_side = new(preview_icon, dir = WEST)
-
-	qdel(face_s)
-	qdel(underwear_s)
-	qdel(undershirt_s)
-	qdel(socks_s)
-	qdel(clothes_s)
+				preview_icon = icon('icons/mob/robots.dmi', "robot", SOUTH)
+				preview_icon.Scale(64, 64)
+				return
+
+	// Set up the dummy for its photoshoot
+	var/mob/living/carbon/human/dummy/mannequin = new()
+	mannequin.status_flags |= GODMODE // Why not?
+	copy_to(mannequin)
+
+	// Determine what job is marked as 'High' priority, and dress them up as such.
+	var/datum/job/previewJob
+	var/highRankFlag = job_support_high | job_medsci_high | job_engsec_high | job_karma_high
+
+	if(job_support_low & SUPPORT)
+		previewJob = job_master.GetJob("Civilian")
+	else if(highRankFlag)
+		var/highDeptFlag
+		if(job_support_high)
+			highDeptFlag = SUPPORT
+		else if(job_medsci_high)
+			highDeptFlag = MEDSCI
+		else if(job_engsec_high)
+			highDeptFlag = ENGSEC
+		else if(job_karma_high)
+			highDeptFlag = KARMA
+
+		for(var/datum/job/job in job_master.occupations)
+			if(job.flag == highRankFlag && job.department_flag == highDeptFlag)
+				previewJob = job
+				break
+
+	if(previewJob)
+		mannequin.job = previewJob.title
+		previewJob.equip(mannequin, TRUE, alt_title = GetPlayerAltTitle(previewJob))
+
+	CHECK_TICK
+	preview_icon = icon('icons/effects/effects.dmi', "nothing")
+	preview_icon.Scale(48+32, 16+32)
+	CHECK_TICK
+	mannequin.setDir(NORTH)
+
+
+	var/icon/stamp = getFlatIcon(mannequin)
+	CHECK_TICK
+	preview_icon.Blend(stamp, ICON_OVERLAY, 25, 17)
+	CHECK_TICK
+	mannequin.setDir(WEST)
+	stamp = getFlatIcon(mannequin)
+	CHECK_TICK
+	preview_icon.Blend(stamp, ICON_OVERLAY, 1, 9)
+	CHECK_TICK
+	mannequin.setDir(SOUTH)
+	stamp = getFlatIcon(mannequin)

```

Comment:
Hmm, even a single refresh results in a 20 CPU -> 50 CPU load on my computer. A cooldown or some other mitigation of this severe load is necessary.

Reply:
I think that what check tick was for, suggestions?

Reply:
Yeah, didn't they just add CD?

Reply:
Yeah, they did.

Reply:
Oof ouch owie... :skull_and_crossbones:

We'd need one hell of a cooldown then.

Reply:
512 may resolve these issues with the icon performance improvements. As such, this may need to wait until we full time switch.